.github/workflows/truffleruby.yml

```diff
@@ -0,0 +1,54 @@
+name: Test Loader.java with TruffleRuby
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  test-loader:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Clone YARP
+      uses: actions/checkout@v3
+      with:
+        path: yarp
+    - name: Setup system Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: 3.2
+        bundler-cache: true
+        working-directory: yarp
+
+    - run: mkdir truffleruby-ws
+    - name: Clone TruffleRuby
+      uses: actions/checkout@v2
+      with:
+        repository: oracle/truffleruby
+        path: truffleruby-ws/truffleruby
+    - name: Setup jt
+      run: echo "$PWD/truffleruby-ws/truffleruby/bin" >> $GITHUB_PATH
+
+    - name: Import latest YARP in TruffleRuby
+      run: tool/import-yarp.sh
+      working-directory: truffleruby-ws/truffleruby
+
+    - name: Install JVMCI
+      run: jt install jvmci
+    - name: Set JAVA_HOME
+      run: echo "JAVA_HOME=$(jt install jvmci)" >> $GITHUB_ENV
+    - name: Build TruffleRuby
```

Comment:
> Is there a way we could check the Java loader without building truffle?

This can be done and it should be the fastest but it will only verify that the files themselves are valid compilable Java.  Personally, at this stage, I am happy with that amount of verification.

> Would JRuby build faster? Could we do this as a cron job at some regular interval?

JRuby currently is ~2 minutes for the loading of JRuby proper: https://github.com/jruby/jruby/actions/runs/4683992966/jobs/8299649052. 

I think one motivation for @eregon in this PR is it will fully test that any changes in YARP will continue to properly run in TR.  So even if this is not here now it is probably needed in the future.  The other mechanism could be for our respective projects to run the equivalent of this nightly (or some period) but then any breaking commits in YARP will not get noticed until later.

Reply:
> Could we cache this build to speed it up in any way?

I think it's not worth the complexity and it's not really feasible because we want to test the latest version of YARP, by copying files in TruffleRuby and build TruffleRuby with those files.